pkg/virt-launcher/virtwrap/network/podnic.go

```diff
@@ -35,17 +35,51 @@ import (
 )
 
 type podNIC struct {
-	vmi              *v1.VirtualMachineInstance
-	podInterfaceName string
-	launcherPID      *int
-	vmiSpecIface     *v1.Interface
-	vmiSpecNetwork   *v1.Network
-	handler          netdriver.NetworkHandler
-	cacheFactory     cache.InterfaceCacheFactory
-	dhcpConfigurator *dhcpconfigurator.Configurator
+	vmi               *v1.VirtualMachineInstance
+	podInterfaceName  string
+	launcherPID       *int
+	vmiSpecIface      *v1.Interface
+	vmiSpecNetwork    *v1.Network
+	handler           netdriver.NetworkHandler
+	cacheFactory      cache.InterfaceCacheFactory
+	dhcpConfigurator  *dhcpconfigurator.Configurator
+	infraConfigurator infraconfigurators.PodNetworkInfraConfigurator
 }
 
 func newPodNIC(vmi *v1.VirtualMachineInstance, network *v1.Network, handler netdriver.NetworkHandler, cacheFactory cache.InterfaceCacheFactory, launcherPID *int) (*podNIC, error) {
+	podnic, err := newPodNICWithWithoutInfraConfigurator(vmi, network, handler, cacheFactory, launcherPID)
+	if err != nil {
+		return nil, err
+	}
+	if podnic.vmiSpecIface.Bridge != nil {
+		if podnic.launcherPID != nil {
```

Comment:
When the constructor es called at virt-launcher before running Phase2 there is no launcherPID.

Reply:
Ok, also I will assert if launcherPID is nil at newPodNIC

Reply:
I'm a bit confused. I couldn't find any code invoking `newPodNICWithoutInfraConfigurator`, besides `newPodNIC`.
It means that both `Phase1` and `Phase2` invoke `newPodNIC`.
`Phase2` doesn't need `infraConfigurator`, I understand that it doesn't get one because its `launcherPID` is nil.
But is it not readable. I you won't say, I would never know `phase2` doesn't have `launcherPID` set in this stage and would just not understand why this `if` statement is here. And for sure won't understand that `phase2` doesn't have `infraConfigurator`.

I believe that at least in this stage (in the future we should split `podNIC`), `network.go` should invoke `newPodNIC` or  `newPodNICWithoutInfraConfigurator` depending on whether it is `Phase1` or `Phase2`.

Reply:
Fair enough, let me call the proper constructor at VMNetworkConfigurator and also assert error at NewPodNIC if there is no launcherPID so we don't have to check it.